extensions/wikia/ArticleVideo/scripts/ooyala-player.js

```diff
@@ -39,14 +42,25 @@ define('ooyala-player', ['wikia.browserDetect'], function (browserDetect) {
 	 * @returns {void}
 	 */
 	OoyalaHTML5Player.prototype.onCreate = function (player) {
-		var messageBus = player.mb,
+		var adTrackingParams = this.params.adTrackingParams,
+			messageBus = player.mb,
 			self = this;
 
+		if (ooyalaTracker && adTrackingParams) {
+			ooyalaTracker.register(player, adTrackingParams);
+		}
+
 		this.onPlayerCreate(player);
 
 		messageBus.subscribe(window.OO.EVENTS.PLAYBACK_READY, 'ui-update', function () {
 			self.onPlaybackReady();
 		});
+
+		messageBus.subscribe(window.OO.EVENTS.PLAYED, 'ads', function () {
+			// Reset line item id and creative id after the video is finished
```

Comment:
What about midroll/postroll? Does it work fine?

Reply:
mid and post roll will set its own Id and player will start reporting this ID into kikimora.